backend/endpoints/rooms.py

```diff
@@ -127,11 +127,17 @@ async def remove_member(
             detail="user not a member of the room",
         )
 
-    if admin_id not in room_data["room_members"]:
+    if admin_id is not None and admin_id not in room_data["room_members"]:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail="admin id specified not a member of the room",
         )
+    
+    if admin_id is not None and member_id == admin_id:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="cannot remove yourself"
+        )
```

Comment:
lines 136 - 140 is completely unnecessary IMO. The endpoint already does what its supposed to without it..